### PR TITLE
feat: add java indent

### PR DESCRIPTION
**Pull request by elianiva**

### CODE REVIEW

**Comment by kyazdani42** [comment]
@p00f could you check that this is right ?

**Comment by p00f** [comment]
sure just a sec

**Comment by p00f** [comment]
for argument list i couldn't find an authoritative style guide but shouldn't it be in one line?
image
The rest is nice!

**Comment by elianiva** [comment]
I don't know about that either :laughing:
I just add the indent query because I recently started writing java at my school so I still don't know much about it.

**Comment by p00f** [comment]
this is better than 
image
though, so I'm fine

**Review by p00f — APPROVED** [review]

File: queries/java/indents.scm [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[
+  (class_declaration)
+  (class_body)
+  (constructor_declaration)
+  (constructor_body)
+  (block)
+  (switch_block)
+  (array_initializer)
+  (argument_list)
```
Comment (p00f) [inline queries/java/indents.scm:9]:
```suggestion
  (argument_list)
  (formal_parameters)
```
Indentation of formal_parameters behaves the same way (not ideal that is), but better than nothing

**Review by p00f — CHANGES_REQUESTED** [review]

File: queries/java/indents.scm (hunk 1 above)
Comment (p00f) [reply queries/java/indents.scm:9]: @theHamsta is the cpp maintainer, can you shed light?

**Comment by p00f** [comment]
Same with array_initializer
image

**Review by p00f — APPROVED** [review]

**Comment by elianiva** [comment]
I have a question, not related to indent but still related to Java. Do we want to inject jsdoc for java comments? They both are very similar, but I'm not sure if it's a good idea.

**Review by kyazdani42 — APPROVED** [review]

**Comment by kyazdani42** [comment]
@elianiva i don't know java much, but if they do not use jsdoc, i don't think we should enable injection for it :)

**Comment by elianiva** [comment]
That makes sense, I'll just do it locally then.

**Comment by theHamsta** [comment]
Maybe this could work. Jsdoc and javadoc pretty much look the same.

**Pull request merged**